routes/users.js

```diff
@@ -1,10 +1,143 @@
+      if(err) res.json({err:'token invalid'});
+      else{
+        req.decoded = decoded;
+        next();
+      }
+    })
+  }else{
+    if((req.originalUrl === '/users/newUser')||(req.originalUrl === '/users/loginUser')) {
+      next();
+    }
+    else
+    res.json({err:"no token provided cann't access to this page" });
+  }
+});
+
+/*******************************************************registraition************************************* */
+router.post('/newUser', function(req, res) {
+  User.count({email: req.body.email}, function(err, c) {
+    if(c === 0){
+      User.count({username: req.body.username}, function(err, c) {
+        if(c === 0){
+          if((req.body.username.trim() == '')||(req.body.password == '')||(req.body.email == '')||
+              (req.body.username == null)||(req.body.password == null)||(req.body.email == null))
+                  res.json({success: false, message: 'check email username and password'});
+          else{
+            let myUser = new User();
+            myUser.username = req.body.username;
+            myUser.email = req.body.email;
+            myUser.password = req.body.password;
+            User.create(myUser)
+                .then(function(dbUser) {
+                  res.json({success: 'success', user: dbUser});
+                 }).catch(function(err) {
+                   console.log(err);
+                      res.json({err: 'can not add to DB'});
+                 });
+            }
+          }else res.json({err: 'this username is already exist!!!'});
+        }).catch(function(err){
+            res.json({err: 'there is som isue in database please call to support team'});
+      });
+    }else res.json({err: 'this email is already exist!!!'});
+  });
+});
+
+
+/*******************************************************login************************************* */
+let validPassword;
+router.post('/loginUser', function(req, res) {
+  User.findOne({email: req.body.emaillogin}).select('username password email').exec(function(err, user){
+    if(err) throw err;
+    console.log(user);
+    if(!user) res.json({err : 'could not authenticate username'});
+    else{
+      const selectedUser = new User(user);
+  
+      if(req.body.passwordlogin)  validPassword = selectedUser.comparePasword(req.body.passwordlogin);
+      else res.json({err : 'No password provided'});
+      if(!validPassword) res.json({err : 'could not authenticate password'});
+      else{
+        let token = jwt.sign({username: User.username, email: User.email}, secret,{expiresIn: '24h'});
+      
+      res.json({token: token, user: user});        
+        // res.json({token: token});
+        // console.log(__dirname + '*******' + selectedUser.password + '--------'+selectedUser.email);
+        //  Inventory.find({/*itemSeller: req.body._id*/}).then(inventory => {
+        //   res.render('index', { inventory: inventory });
+        // });
+      }
+    }
+  });
+});
+
+
```

Comment:
You cannot mix /inventory path in users.js.   We have two different paths '/' and '/users'.   inventory belongs to "/' path.  if you use '/inventory/:user' route here, it is actually changed to "/users/inventory/:user."

Reply:
so how should use router.use middleware to check token of user in index.js route?